src/sql/workbench/services/notebook/browser/models/notebookModel.ts

```diff
@@ -449,6 +433,36 @@ export class NotebookModel extends Disposable implements INotebookModel {
 		}
 	}
 
+	private loadContentMetadata(metadata: any): void {
+		this._savedKernelInfo = metadata.kernelspec;
+		this._defaultLanguageInfo = metadata.language_info;
+		// If language info was serialized in the notebook, attempt to use that to decrease time
+		// required until colorization occurs
+		if (this._defaultLanguageInfo) {
+			this.updateLanguageInfo(this._defaultLanguageInfo);
+		}
+		this._savedConnectionName = metadata.connection_name;
+		this._multiConnection = metadata.multi_connection ? metadata.multi_connection : false;
+
+		//Telemetry of loading notebook
+		if (metadata.azdata_notebook_guid && metadata.azdata_notebook_guid.length === 36) {
+			//Verify if it is actual GUID and then send it to the telemetry
+			let regex = new RegExp('(\{){0,1}[0-9a-fA-F]{8}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{12}(\}){0,1}');
+			if (regex.test(metadata.azdata_notebook_guid)) {
+				this.adstelemetryService.createActionEvent(TelemetryKeys.TelemetryView.Notebook, TelemetryKeys.TelemetryAction.Open)
+					.withAdditionalProperties({ azdata_notebook_guid: metadata.azdata_notebook_guid })
+					.send();
+			}
+		}
+		Object.keys(metadata).forEach(key => {
+			let expectedKeys = ['kernelspec', 'language_info', 'tags', 'connection_name', 'multi_connection'];
```

Comment:
Can you work on figuring out a better way to do this? It's going to be annoying to have to remember to update this every time we add a new field. There isn't a native way to iterate over the keys of an interface AFAIK unfortunately - but maybe there's some other way we can do that.

At the very least this array should be defined next to the interface so it's easier to notice and update when modifying the interface.

Reply:
No way to get the keys of an interface during runtime but I created an object of type Required<nb.INotebookMetadata> so that when the interface is modified, the object will also have to be updated.